integr8ly/integreatly-operator
# Create alertmanger secrets for OO and AMO

# Issue link

https://issues.redhat.com/browse/MGDAPI-1879

# What
Create the AlertmanagerConfig secret in the Observability operator namespace

# Verification steps
- Deploy the branch locally 
- Manually approve the install plans for Prometheus and Grafana
- Go to the route for Alertmanager from the `redhat-rhoam-observability` NS
- Go to Status and confirm that the config is present 
- Go to the route for Alertmanger  from the `redhat-rhoam-middleware-monitoring-operator` NS
- Go to Status and confirm that the config is present

![image](https://user-images.githubusercontent.com/16667688/132517961-a0d70669-d225-46e7-b276-c2f5f081fd35.png)


```yaml
global:
  resolve_timeout: 5m
  http_config: {}
  smtp_from: noreply-alert@devshift.org
  smtp_hello: localhost
  smtp_smarthost: smtp.example.com:587
  smtp_require_tls: true
  pagerduty_url: https://events.pagerduty.com/v2/enqueue
  opsgenie_api_url: https://api.opsgenie.com/
  wechat_api_url: https://qyapi.weixin.qq.com/cgi-bin/
  victorops_api_url: https://alert.victorops.com/integrations/generic/20131114/alert/
route:
  receiver: default
  routes:
  - receiver: critical
    match:
      severity: critical
  - receiver: deadmansswitch
    match:
      alertname: DeadMansSwitch
    repeat_interval: 5m
  - receiver: BUandCustomer
    match:
      alertname: RHOAMApiUsageLevel1ThresholdExceeded
  - receiver: BUandCustomer
    match:
      alertname: RHOAMApiUsageLevel2ThresholdExceeded
  - receiver: SRECustomerBU
    match:
      alertname: RHOAMApiUsageLevel3ThresholdExceeded
  - receiver: BU
    match_re:
      alertname: RHOAMApiUsageSoftLimitReachedTier[0-9]+
  - receiver: BUandCustomer
    match:
      alertname: RHOAMApiUsageOverLimit
  - receiver: SRECustomerBU
    match:
      alertname: RHOAMApiUsageRejectedRequestsMismatch
  group_wait: 30s
  group_interval: 5m
  repeat_interval: 12h
inhibit_rules:
- source_match:
    alertname: JobRunningTimeExceeded
    severity: warning
  target_match:
    alertname: JobRunningTimeExceeded
    severity: warning
  equal:
  - alertname
  - job
  - label_cronjob_name
receivers:
- name: default
  email_configs:
  - send_resolved: true
    to: noreply-test@rhmi-redhat.com
    from: noreply-alert@devshift.org
    hello: localhost
    smarthost: smtp.example.com:587
    headers:
      From: noreply-alert@devshift.org
      Subject: '{{template "email.integreatly.subject" . }}'
      To: noreply-test@rhmi-redhat.com
    html: '{{ template "email.integreatly.html" . }}'
    require_tls: true
- name: critical
  email_configs:
  - send_resolved: true
    to: noreply-test@rhmi-redhat.com
    from: noreply-alert@devshift.org
    hello: localhost
    smarthost: smtp.example.com:587
    headers:
      From: noreply-alert@devshift.org
      Subject: '{{template "email.integreatly.subject" . }}'
      To: noreply-test@rhmi-redhat.com
    html: '{{ template "email.integreatly.html" . }}'
    require_tls: true
  pagerduty_configs:
  - send_resolved: true
    http_config: {}
    service_key: <secret>
    url: https://events.pagerduty.com/v2/enqueue
    client: '{{ template "pagerduty.default.client" . }}'
    client_url: '{{ template "pagerduty.default.clientURL" . }}'
    description: '{{template "email.integreatly.subject" . }}'
    details:
      cluster_ID: 9eb01722-d7e0-40c2-9c3d-3c0f4de5ba39
      cluster_name: mw-collab-byoc-6qvjc
      console: https://console-openshift-console.apps.mw-collab-byoc.2dv3.s1.devshift.org
      firing: '{{ template "pagerduty.default.instances" .Alerts.Firing }}'
      num_firing: '{{ .Alerts.Firing | len }}'
      num_resolved: '{{ .Alerts.Resolved | len }}'
      resolved: '{{ template "pagerduty.default.instances" .Alerts.Resolved }}'
- name: deadmansswitch
  webhook_configs:
  - send_resolved: true
    http_config: {}
    url: https://dms.example.com
    max_alerts: 0
- name: BU
  email_configs:
  - send_resolved: true
    to: noreply-test@rhmi-redhat.com
    from: noreply-alert@devshift.org
    hello: localhost
    smarthost: smtp.example.com:587
    headers:
      From: noreply-alert@devshift.org
      Subject: '{{template "email.integreatly.subject" . }}'
      To: noreply-test@rhmi-redhat.com
    html: '{{ template "email.integreatly.html" . }}'
    require_tls: true
- name: BUandCustomer
  email_configs:
  - send_resolved: true
    to: noreply-test@rhmi-redhat.com, noreply@kafka-alertmanager-redhat-rhoam-observability-operator.apps.mw-collab-byoc.2dv3.s1.devshift.org
    from: noreply-alert@devshift.org
    hello: localhost
    smarthost: smtp.example.com:587
    headers:
      From: noreply-alert@devshift.org
      Subject: '{{template "email.integreatly.subject" . }}'
      To: noreply-test@rhmi-redhat.com, noreply@kafka-alertmanager-redhat-rhoam-observability-operator.apps.mw-collab-byoc.2dv3.s1.devshift.org
    html: '{{ template "email.integreatly.html" . }}'
    require_tls: true
- name: SRECustomerBU
  email_configs:
  - send_resolved: true
    to: noreply-test@rhmi-redhat.com, noreply-test@rhmi-redhat.com, noreply@kafka-alertmanager-redhat-rhoam-observability-operator.apps.mw-collab-byoc.2dv3.s1.devshift.org
    from: noreply-alert@devshift.org
    hello: localhost
    smarthost: smtp.example.com:587
    headers:
      From: noreply-alert@devshift.org
      Subject: '{{template "email.integreatly.subject" . }}'
      To: noreply-test@rhmi-redhat.com, noreply-test@rhmi-redhat.com, noreply@kafka-alertmanager-redhat-rhoam-observability-operator.apps.mw-collab-byoc.2dv3.s1.devshift.org
    html: '{{ template "email.integreatly.html" . }}'
    require_tls: true
templates:
- /etc/alertmanager/config/alertmanager-email-config.tmpl
```

# Issues
`http_config: {}`  is populated when the same config template is applied with AMO 
```yaml
http_config:            
   follow_redirects: true
```

`routes` have an extra field `continue: false`  when deployed with AMO which is not present in the OO config   e.g.
```yaml
route:            
    continue: false            
    group_interval: 5m            
    group_wait: 30s            
    receiver: default            
    repeat_interval: 12h            
        routes:            
        - continue: false            
          match:            
              severity: critical            
          receiver: critical
```
